### ISSUE TITLE
Allow coroutines to be used in dask.bag operations

### ISSUE THREAD

**Issue by ianliu** [dask/dask#8546]
Would be nice if dask allowed to pass coroutines directly to dask bag operations, such as map, filter, foldby, etc.

Seems like https://github.com/dask/distributed/pull/5151 added the ability to submit coroutines on the client object.

I've submitted this request here first: https://dask.discourse.group/t/is-it-possible-to-map-coroutines-in-a-dask-bag/174

**Comment by scharlottej13** [comment]
Thanks @ianliu for opening this issue! Would you mind expanding a little bit on your use case for this? @jrbourbeau or @jsignell do you know if this has come up before?

**Comment by jsignell** [comment]
Yeah I think this has come up before but I can't find a reference.

**Comment by jsignell** [comment]
I think I was thinking about an issue where the user wanted to pass iterables of futures to `from_delayed` so I am no longer sure if there has been a request to pass coroutines
